sql/session.go

```diff
@@ -51,13 +53,25 @@ func (s *BaseSession) Address() string { return s.addr }
 func (s *BaseSession) Set(key string, typ Type, value interface{}) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+
+	key = strings.TrimLeft(key, "@")
```

Comment:
so, all variables are set as session level? what happens when the user disconnects? per example `autocommit=1` will disappear?

Reply:
autocommit is a session variable, isn't it?

Reply:
Here what I only changed is checking if we have session prefix and remove them before set. Because right now we don't have Qualifiers in `SetExpr`, so you do sth, like:
```
SET @autocommit = 1
SELECT @@session.autocommit
```
we'll set in `s.config["autocommit"]=1` but for select we'll try to get `s.config["session.autocommit"]` what is not there.

Reply:
that's why the `session.` should be removed in the select

Reply:
ok, makes sense. I've moved that logic from session.Get/Set to 
`analyzer.resolveColumns` and to `plan.(*Set)RowIter`

Reply:
I had to fix `TestResolveColumnsSession`. I think right now it wasn't right (correct me if I'm wrong):
it looked like:
```
node := plan.NewProject(
		[]sql.Expression{
			expression.NewUnresolvedColumn("@@foo_bar"),
			expression.NewUnresolvedColumn("@@bar_baz"),
		},
		plan.NewResolvedTable(dualTable),
	)

	result, err := resolveColumns(ctx, NewDefault(nil), node)
	require.NoError(err)

	expected := plan.NewProject(
		[]sql.Expression{
			expression.NewGetSessionField("@@foo_bar", sql.Int64, int64(42)), // <-- @@-
			expression.NewGetSessionField("@@bar_baz", sql.Null, nil), // <-- @@
		},
		plan.NewResolvedTable(dualTable),
	)
```

I assumed that in session we don't keep fields prefixed by `@`, because:
- if you take a look into session config:
```
func defaultSessionConfig() map[string]typedValue {
	return map[string]typedValue{
		"auto_increment_increment": typedValue{Int64, int64(1)},
		"time_zone":                typedValue{Text, time.Local.String()},
		"system_time_zone":         typedValue{Text, time.Local.String()},
		"max_allowed_packet":       typedValue{Int32, math.MaxInt32},
		"sql_mode":                 typedValue{Text, ""},
	}
}
```
and also `func (f *GetSessionField) String() string { return "@@" + f.name }` already prefixes with "@".